mantidproject/mantid
# Update ILL reflectometry loader and D17 and Figaro IDF

This PR updates [`LoadILLReflectometry`](http://docs.mantidproject.org/nightly/algorithms/LoadILLReflectometry-v1.html) to correctly load current D17 NeXus files. The file format for the Figaro instrument is still under development so the loader cannot open those files yet.

Additionally, the IDFs of D17 and Figaro were updated.

**To test:**

The output of the loader has been validated against the legacy LAMP software at ILL. Therefore a code review and the standard try-to-break-it should suffice. A test script might look like this (add your Mantid build path to the appropriate position):
```python
mantid_data_dir = '<tester input needed here>/ExternalData/Testing/Data/UnitTest/ILL/D17/'
mantid.config.appendDataSearchDir(mantid_data_dir)

# Load direct beam data, note the extra optional output
LoadILLReflectometry(Filename='317369', OutputWorkspace='DB', OutputBeamPosition='beam_position', XUnit='TimeOfFlight')
# Load reflected beam. Use direct beam information to calibrate the detector angle.
LoadILLReflectometry(Filename='317370', OutputWorkspace='RB', BeamPosition='beam_position', XUnit='TimeOfFlight')
```

Fixes #18235 and #19029.

**Release Notes** 

The release notes were updated accordingly.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
